### PR TITLE
LIP0065: hash events before inserting them into tree

### PR DESCRIPTION
**Pull request by ricott1**
Update the specifications of the events tree: set the leaf value to the _hash_ of the serialized event.

### CODE REVIEW

**Review by shuse2 — APPROVED** [review]

File: proposals/lip-0065.md [hunk 1]
```diff
@@ -206,12 +206,12 @@ For events emitted during the command execution, modules specify whether the eve
...
+Events are inserted into the events tree once for each entry in the event `topics` array. The leaf key is the concatenation of the hash of the element in the topics array, the event `index` converted to a binary string of length `EVENT_INDEX_LENGTH_BITS` and concatenated with the topic index of `TOPIC_INDEX_LENGTH_BITS` bits (corresponding to the entry in the `topics` array), for a total of `TOTAL_INDEX_LENGTH_BYTES` bytes. The leaf value is the SHA-256 hash of the serialization of the event properties using the `eventSchema` given above.
 
 In summary, for each event one key-value pair is added to the events tree for each entry `topic` in the associated `topics` array, where:
 
-* `key = SHA-256(topic)[:TOPIC_HASH_LENGTH_BYTES] + ((event index << TOPIC_INDEX_LENGTH_BITS) + topic index).to_bytes(TOTAL_INDEX_LENGTH_BYTES, byteorder="big")`
-* `value` = serialization of event according to `eventSchema`.
+* `key = sha256(topic)[:TOPIC_HASH_LENGTH_BYTES] + ((event index << TOPIC_INDEX_LENGTH_BITS) + topic index).to_bytes(TOTAL_INDEX_LENGTH_BYTES, byteorder="big")`
```
Comment (gkoumout) [inline proposals/lip-0065.md:213]: In the constants table and in `eventID()` the hash it is denoted by `SHA-256()` and here `sha256()`. I suggest using the same notation accross the whole LIP for consistency.

**Review by gkoumout — APPROVED** [review]
Ok from my side, just left a comment on notation consistency.

File: proposals/lip-0065.md (hunk 1 above)
Comment (ricott1) [reply proposals/lip-0065.md:213]: fixed it in the code, I left it as `SHA-256` in the text (we do the same in other LIPs)

File: proposals/lip-0065.md [hunk 2]
```diff
@@ -98,7 +98,7 @@ To summarize, we chose sparse Merkle trees because:
 
 | Name | Type | Value | Description |
 |------|------|-------|-------------|
-| `MAX_EVENT_SIZE_BYTES` | integer | `1024` | Maximum size in bytes of a serialized event. |
+| `MAX_EVENT_SIZE_BYTES` | integer | `16*1024` | Maximum size in bytes of a serialized event. |
```
Comment (gkoumout) [inline proposals/lip-0065.md:101]: I have missed the background behind this change, but still fine from my side

**Review by gkoumout — APPROVED** [review]

File: proposals/lip-0065.md (hunk 2 above)
Comment (ricott1) [reply proposals/lip-0065.md:101]: is to be able to emit complete ccms into events. I added a generous buffer (also 16 is a round number).
